### DISCUSSION TITLE
Make cached array from ObservableMap values

### DISCUSSION THREAD

**Discussion by Avircut** [mobxjs/mobx#3933] [Q&A]
Hi, I'm trying to get array from ObservableMap literally with 
`  get providersMap(): ObservableMap<number, ProviderWithFoldersAndResources> {
    return this.providers.items;
  }
  get providersArray(): ProviderWithFoldersAndResources[] {
    return Array.from(this.providers.items.values());
  }`
My store is autoObservable and I tried to add @computed decorator to these getters, but if providerMap is cached, providersArray changes on every render. 
Autorun on providers.items.values() also works fine.
Please tell me what's wrong?

**Comment by Avircut** [comment]
Also tried to override providers computed with keepAlive:
`    makeAutoObservable(this, {
      providersArray: computed({ keepAlive: true }),
    });`
But that didn't work too

**Reply by Avircut** [reply]
Okay, with trace I found out that array actually was changed all the times and autorun was fine coz array object was the same, and just values changed, but how to get the same array from ObservableMap values?

**Comment by mweststrate** [comment]
Please provide a small code sandbox with what you are trying to do,
including how you consume the outputs. It's a bit hard to follow from the
description but from the question it seems your mental model is not
entirely accurate here, but it is hard to pinpoint from just an abstract
description 😅
